include/pybind11/pytypes.h

```diff
@@ -1588,7 +1588,7 @@ class capsule : public object {
                 }
                 pybind11_fail("Unable to get capsule context");
             }
-            void *ptr = PyCapsule_GetPointer(o, nullptr);
+            void *ptr = PyCapsule_GetPointer(o, PyCapsule_GetName(o));
```

Comment:
PyCapsule_GetName can throw exceptins so the error handling is a little more subtle here. I'll go and fix it when I have time.

Reply:
Oh, before I wasn't aware of the fact that we need to pass the name here. I think I misunderstood.

Now looking around, I see there is already similar code in `get_pointer`.
That uses `this->name();` without checking error conditions as documented here:

https://docs.python.org/3/c-api/capsule.html#c.PyCapsule_GetName

This looks like an existing bug.

Current thinking:
* add error handling to the `name()` implementation.
* re-use the thus indirectly fixed-up `get_pointer` implementation here and below.

Reply:
I just pushed a change to add error handling in `name()` method.